ui/src/components/modals/DeleteModelModal.js

```diff
@@ -0,0 +1,132 @@
+  // const [inactiveEndpoint, setInactiveEndpoint] = useState([])
+  const [deleteConfirmation, setDeleteConfirmation] = useState('')
+
+  const [{ isLoading, isLoaded }, deleteModel] = useMerlinApi(
+    `projects/${model.project_id}/models/${model.id}`,
+    { method: "DELETE", addToast: true, mock: mocks.noBody },
+    {},
+    false
+  );
+
+  const[versions] = useMerlinApi(
+    `/models/${model.id}/versions`,
+    {}
+  )
+
+  useEffect(() => {
+    if (isLoaded) {
+      setDeleteConfirmation("")
+      closeModal();
+      callback();
+    }
+  }, [isLoaded, closeModal, callback]);
+
+  useEffect(() => {
+    const activeItem = [];
+    // const inactiveItem = [];
+    if (versions.isLoaded){
+      for (const item of versions.data){
+        const activeEnd = item.endpoints.filter(
+          (endpoint) => ["pending", "running", "serving"].includes(endpoint.status)
+        );
+        if (activeEnd.length > 0){
+          activeItem.push(activeEnd)
+        } else{
+          // inactiveItem.push(item)
+          continue
+        }
+      }
+    }
+    setActiveEndpoint(activeItem)
+    // setInactiveEndpoint(inactiveItem) 
+  }, [versions])
+
+  return (
+    <EuiOverlayMask>
+      <EuiConfirmModal
+        title="Delete Model"
+        onCancel={closeModal}
+        onConfirm={deleteModel}
+        cancelButtonText="Cancel"
+        confirmButtonText="Delete"
+        buttonColor="danger"
+        confirmButtonDisabled={deleteConfirmation !== model.name || activeEndpoint.length > 0}>
+        {isLoading && (
+          <EuiProgress
+            size="xs"
+            color="accent"
+            className="euiProgress-beforePre"
+          />
+        )}
+        {versions.isLoading ? (
+            <EuiProgress
+            size="xs"
+            color="accent"
+            className="euiProgress-beforePre"
+          />
+        ) : (
+        <div>
+          {model.endpoints.length > 0 ? (
+            <span>
+              You cannot delete this Model because there are <b> {model.endpoints.length} Endpoints</b> using this model. 
+              <br/> <br/> If you still wish to delete this model, please <b>Stop Serving</b> Endpoints that use this model. <br/>
+            </span>        
+          ) : (
+            <div>
+            {activeEndpoint.length > 0 ? (
+              <span>
+                You cannot delete this Model because there are <b> {activeEndpoint.length} Endpoints</b> currently using the model version associated with it. 
+                <br/> <br/> If you still wish to delete this model, please <b>Undeploy</b> Endpoints that use this version. <br/>
+              </span>
+            ) : (
+              <div>
+                  You are about to delete model <b>{model.name}</b>. This action cannot be undone. 
+                  <br/> <br/> Please note that all the related entity of this model (<b>Model version</b> and <b>Model Version Endpoint)</b> will be <b>deleted</b>.
```

Comment:
Nit: all the related `entity` -> all the related `entities` (plural is more accurate).